docs/vhost-user-blk-testing.md

```diff
@@ -0,0 +1,102 @@
+# How to test Vhost-user block with SPDK
+
+The purpose of this document is to illustrate how to test vhost-user-blk in cloud-hypervisor with SPDK as the backend.
+
+## Framework
+
+It's a simple test to validate the block read/write between VM and block backend.
+```
+             +----+----------+          +-------------+-----------+
+             |    |          |          |             |           |
+             |    |vhost-user|----------| vhost-user  |    dpdk   |
+             |    |blk device|          | port 1      |           |
+             |    |          |          |             |           |
+             |    +----------+          +-------------+-----------+
+             |               |          |                         |
+             |      vm       |          |           spdk          |
+             |               |          |                         |
+          +--+----------------------------------------------------+--+
+          |  |                  hugepages                         |  |
+          |  +----------------------------------------------------+  |
+          |                                                          |
+          |                       host                               |
+          |                                                          |
+          +----------------------------------------------------------+
+```
+## Prerequisites
+
+Prior to running the test, the following steps need to be performed.
+- Enable hugepages
+- Install SPDK
+
+Here are some good references for detailing them.
+- spdk
+	* https://spdk.io/doc/
+
+## Test environment
+
+The below test environment is based on ubuntu release, as for other system, please check related document.
+The test runs with multiple queue (MQ) support enabled, using 4 queues defined for both SPDK and the virtual machine. Here are the details on how the test can be run.
+
+# Host linux cmdline setting
+Add "default_hugepagesz=1G hugepagesz=1G hugepages=2" into /boot/grub/grub.cfg
+
+# Download the SPDK code
+git clone https://github.com/spdk/spdk
+cd spdk
+git submodule update --init
+
+# create the build dep
+./scripts/pkgdep.sh
+
+# build spdk
+./configure
+make
+
+# set the SPDk environment
+sudo sysctl -w vm.nr_hugepages=2
+sudo mount -t hugetlbfs -o pagesize=1G none /dev/hugepages
+sudo HUGEMEM=2048 scripts/setup.sh
+sudo ./app/vhost/vhost -S /var/tmp -s 1024 -m 0x3 &
+
+# create 512M block device
+sudo scripts/rpc.py bdev_malloc_create 512 512 -b Malloc0
+sudo scripts/rpc.py vhost_create_blk_controller --cpumask 0x1 vhost.1 Malloc0
+
+_Launch the VM_
+
+VMs run in client mode. They connect to the socket created by the `dpdkvhostuser` backend.
```

Comment:
Is the backend really called `dpdkvhostuser` ?

Reply:
Glad to help. But I am busy with other tasks. So if it is not so urgent, I can help with it. :)

Reply:
Sorry, this should be "`dpdkvhostuser` in the SPDK backend", thanks.